hatchery/nextflow_test.go

```diff
@@ -49,3 +52,98 @@ runcmd:
 		t.Errorf("The 'generateEcrLoginUserData' function should have returned the expected output '%v', but it returned: '%v'", expectedOutput, userData)
 	}
 }
+
+func TestGetNextflowInstanceAmi(t *testing.T) {
+	defer SetupAndTeardownTest()()
+
+	instanceAmiValue := "instance-ami"
+	instanceAmiBuilderArnValue := "instance-ami-builder-arn"
+	builderLatestAmi := "latest-ami"
+
+	Config.ContainersMap = map[string]Container{
+		"container_with_instance_ami": {
+			NextflowConfig: NextflowConfig{
+				InstanceAmi:           instanceAmiValue,
+				InstanceAmiBuilderArn: "",
+			},
+		},
+		"container_with_instance_ami_builder_arn": {
+			NextflowConfig: NextflowConfig{
+				InstanceAmi:           "",
+				InstanceAmiBuilderArn: instanceAmiBuilderArnValue,
+			},
+		},
+		"container_with_both": {
+			NextflowConfig: NextflowConfig{
+				InstanceAmi:           instanceAmiValue,
+				InstanceAmiBuilderArn: instanceAmiBuilderArnValue,
+			},
+		},
+		"container_with_neither": {
+			NextflowConfig: NextflowConfig{
+				InstanceAmi:           "",
+				InstanceAmiBuilderArn: "",
+			},
+		},
+	}
+
+	// mock the `imagebuilder.ListImagePipelineImages` call to AWS
+	mockedListImagePipelineImages := func(input *imagebuilder.ListImagePipelineImagesInput) (*imagebuilder.ListImagePipelineImagesOutput, error) {
+		// on the 1st call, return an old image and a NextToken to trigger a 2nd call
+		output := imagebuilder.ListImagePipelineImagesOutput{
+			ImageSummaryList: []*imagebuilder.ImageSummary{
+				{
+					DateCreated: aws.String("2023-03-03T00:00:00Z"),
```

Comment:
should this be calculated when the function is run?

Reply:
it's mocking listing images, not creating images, so this should be fine

the mocked function returns 2 images, the 2 dates are hardcoded so we can test the comparison logic, since we want to test that `getNextflowInstanceAmi()` returns the latest image